addons/hr_holidays/hr_holidays.py

```diff
@@ -390,7 +397,7 @@ def holidays_validate(self, cr, uid, ids, context=None):
             if record.holiday_type == 'employee' and record.type == 'remove':
                 meeting_obj = self.pool.get('calendar.event')
                 meeting_vals = {
-                    'name': record.name or _('Leave Request'),
+                    'name': record.display_name,
```

Comment:
Why remove the "or" ? Because of that, the name I give in the form view is automatically replaced by the display_name. Either get the "or" back, or remove the name field from the form view if you don't want people to use it anymore.